### PR TITLE
Branch 0.8

### PR DESCRIPTION
**Pull request by mark-kar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,19 +11,23 @@ extensible to support any kind of data stores.
...
+Criteria     :=  ConjOp Criteria Criteria | RefExpr | Value<Boolean>
+ConjOp       :=  AND | OR
+RefExpr      :=  UnPredOp Ref | BinPredOp Ref Ref | RefColOp Ref Collection<V> | RefRangeOp Ref Range<V> | RegexOp Ref Value<String>
+Ref          :=  Value<T> | Col
+UnPredOp     :=  IS_NULL | IS_NOT_NULL
+BinPredOp    :=  EQ | NEQ | GT | LT | GEQ | LEQ
+RefColOp     :=  IN | NOTIN
+RefRangeOp   :=  BETWEEN | NOTBETWEEN
+RegexOp      :=  LIKE
 ```
```
Comment (mark-kar) [inline README.md:23]: Well, I tried to fix this Backus–Naur form in order to describe actual form of criteria
But now it looks a little bit complicated and actually I think it should be (just look at the formal expressions of postgresql expressions😁)

In my opinion, we could exclude this theoretical part from README and add more practical examples below🙂

What do you think?

**Review by rafafrdz — COMMENTED** [review]
Again, great job! :) I really love

I put again some questions that I asked you in the older PR that I would like to know and to really understand. The rest of suggestion that I wrote you are perfect now.

File: core/src/main/scala/io/github/rafafrdz/criteria4s/core/Ref.scala [hunk 2]
```diff
@@ -1,43 +1,38 @@
...
-
-  trait Bool[D <: CriteriaTag] extends Value[Boolean, D] with Criteria[D] {
-    self =>
-
-    override def value: String = ref.value
+  trait Value[D <: CriteriaTag, +V]      extends Ref[D, V]
+  trait Collection[D <: CriteriaTag, +V] extends Ref[D, Seq[V]]
+  trait Col[D <: CriteriaTag]            extends Ref[D, Column]
+  trait Bool[D <: CriteriaTag]           extends Value[D, Boolean] with Criteria[D]
+  trait Range[D <: CriteriaTag, +V]      extends Ref[D, (V, V)]
```
[the remark this answers is not in the document]
Comment (mark-kar) [reply core/src/main/scala/io/github/rafafrdz/criteria4s/core/Ref.scala:13]: well, it was just for aesthetics, but I think covariance is redundant here, I'll fix it (don't remember why did i do it🤔)

File: README.md (hunk 1 above)
Comment (mark-kar) [reply README.md:23]: okay, I get it, then I'll make it as you say :)

File: core/src/main/scala/io/github/rafafrdz/criteria4s/core/Ref.scala [hunk 3]
```diff
@@ -1,43 +1,38 @@
...
 
+  private[criteria4s] def array[V, D <: CriteriaTag](vs: V*): Collection[D, V] =
+    (show: Show[Seq[V], D]) => show.show(vs)
+
+  private[criteria4s] def range[V, D <: CriteriaTag](left: V, right: V): Range[D, V] =
+    (show: Show[(V, V), D]) => show.show((left, right))
+
 }
+
+class Column(val colName: String) extends AnyVal
```
[the remark this answers is not in the document]
Comment (mark-kar) [reply core/src/main/scala/io/github/rafafrdz/criteria4s/core/Ref.scala:38]: I don't like the idea to use trait here as it should be just new type of string to use implicits for it

I guess the second variant you suggest is the best choice here

**Review by rafafrdz — APPROVED** [review]
LGTM 🚀

**Pull request merged**